### PR TITLE
authn: add task for per-path jwt authentication

### PR DESCRIPTION
**Pull request by yangminzhu**
/cc @diemtvu Could you review this? (It seems I cannot add you to reviewers). Thanks!

### CODE REVIEW

File: content/docs/concepts/security/index.md [hunk 1]
```diff
@@ -444,18 +444,22 @@ mutual TLS implementations.
 
 The `origins:` section defines authentication methods and associated parameters
 supported for origin authentication. Istio only supports JWT origin
-authentication. However, a policy can list multiple JWTs by different issuers.
+authentication. However, a policy can list multiple JWTs by different issuers and be enabled or disabled
+for specific paths.
 Similar to peer authentication, only one of the listed methods must be
 satisfied for the authentication to pass.
```
[the remark this answers is not in the document]
Comment (yangminzhu) [reply content/docs/concepts/security/index.md]: Done.

**Comment by yangminzhu** [comment]
@diemtvu @liminw @geeknoid Could you review this and comment if you think it needs any change? Thanks.

File: content/docs/concepts/security/index.md [hunk 2]
```diff
@@ -444,18 +444,23 @@ mutual TLS implementations.
 
 The `origins:` section defines authentication methods and associated parameters
 supported for origin authentication. Istio only supports JWT origin
-authentication. However, a policy can list multiple JWTs by different issuers.
+authentication. However, a policy can list multiple JWTs by different issuers and be enabled or disabled
+for specific paths. If all JWTs are disabled for a request path, authentication also passes as if
+there is none defined.
```
Comment (liminw) [inline content/docs/concepts/security/index.md]: However ... => You can specify allowed JWT issuers, and enable or disable JWT authentication for a specific path. 

"If all JWTs are disabled for a request path, authentication also passes as if there is none defined." This is confusing. You already said you can disable JWT authentication for a path, isn't that enough?

File: content/docs/concepts/security/index.md [hunk 3]
```diff
@@ -444,18 +444,23 @@ mutual TLS implementations.
...
 Similar to peer authentication, only one of the listed methods must be
 satisfied for the authentication to pass.
 
 The following example policy specifies an `origins:` section for origin
-authentication that accepts JWTs issued by Google:
+authentication that accepts JWTs issued by Google but excludes the path `/health` from the authentication:
```
Comment (liminw) [inline content/docs/concepts/security/index.md]: but excludes ... => JWT authentication for `health` path is disabled.

File: content/docs/tasks/security/authn-policy/index.md [hunk 4]
```diff
@@ -570,6 +570,96 @@ $ for i in `seq 1 10`; do curl --header "Authorization: Bearer $TOKEN" $INGRESS_
 401
 {{< /text >}}
 
+### End-user authentication with per-path requirements
+
+End-user authentication can be enabled or disabled based on request path. This is useful if you want to
+whitelist some paths from End-user authentication, like for paths used for health check or status report.
```
Comment (liminw) [inline content/docs/tasks/security/authn-policy/index.md]: Do not use "whitelist". 

Reword: 
This is useful if you disable authentication for some paths, for example, the path used for health check or status report.

File: content/docs/tasks/security/authn-policy/index.md [hunk 5]
```diff
@@ -570,6 +570,96 @@ $ for i in `seq 1 10`; do curl --header "Authorization: Bearer $TOKEN" $INGRESS_
...
+You can also specify different JWT requirements on different request paths.
+
+#### Whitelist paths from End-user authentication
```
Comment (liminw) [inline content/docs/tasks/security/authn-policy/index.md]: Whitelist ... => Disable End-user authentication for specific paths

Applies to other places as well. Do not call it "whitelist". It is not very clear what it means.

File: content/docs/concepts/security/index.md (hunk 2 above)
Comment (yangminzhu) [reply content/docs/concepts/security/index.md]: That's added per @diemtvu's last comment. It makes it clear the default behavior when all JWT policies are disabled.

**Comment by yangminzhu** [comment]
@liminw @diemtvu @geeknoid All comments addressed, PTAL. Thank you.

File: content/docs/concepts/security/index.md (hunk 2 above)
Comment (liminw) [reply content/docs/concepts/security/index.md]: I am ok if you want to keep this sentence. But I feel it is quite redundant.

**Pull request merged**